x-pack/test/functional/services/ml/job_expanded_details.ts

```diff
@@ -10,13 +10,23 @@ import expect from '@kbn/expect';
 import { FtrProviderContext } from '../../ftr_provider_context';
 import { MlADJobTable } from './job_table';
 
-export function MachineLearningExpandedJobDetailsProvider(
+export function MachineLearningJobExpandedDetailsProvider(
   { getService }: FtrProviderContext,
   jobTable: MlADJobTable
 ) {
   const testSubjects = getService('testSubjects');
+  const find = getService('find');
 
   return {
+    async clickEditAnnotationAction(jobId: string) {
+      await testSubjects.click(jobTable.detailsSelector(jobId, 'euiCollapsedItemActionsButton'));
```

Comment:
Clarifying on my last comment
> and probably also allow to address a specific row in the annotations table (there's the mlAnnotationsTableRow ${id} test subj we could use)

Right now, we're already on the context of on anomaly detection job row. However, in that nested annotations table there might be more than one annotation. Right now, the code just picks the annotations row that's returned first by the find call. Ideally, we would we able to address the annotations row we want to edit.

Reply:
ahh that makes sense.  on it boss

Reply:
https://github.com/elastic/kibana/commit/58efaf9a69678a1085670535a85f257497eb3423